src/functional/iterator/tracing.py

```diff
@@ -148,7 +148,7 @@ def make_node(o):
         return list(make_node(arg) for arg in o)
     if o is None:
         return NoneLiteral()
-    if isinstance(o, iterator.runtime.FundefDispatcher):
+    if hasattr(o, "fun"):
```

Comment:
Change required to support unapplied builtin functions (can be introduced by the new EtaReduction pass).